src/main/java/org/mskcc/oncokb/curation/service/CancerTypeService.java

```diff
@@ -128,6 +142,202 @@ public Optional<CancerType> findByMainTypeAndSubtypeIsNull(String mainType) {
+                    Optional<CancerType> allTumors = cancerTypeRepository.findOneByCodeIgnoreCase(SpecialCancerType.ALL_TUMORS.name());
+                    if (allTumors.isPresent()) {
+                        relevantCancerTypes.add(allTumors.get());
+                    }
+                    break;
+                default:
+                    break;
+            }
+        } else if (direction.equals(RelevantCancerTypeDirection.DOWNWARD)) {
+            Set<CancerType> allOncoTreeTypes = new HashSet<>();
+            List<CancerType> allCancerTypes = cancerTypeRepository.findAll();
+            allOncoTreeTypes.addAll(allCancerTypes);
+
+            List<CancerType> specialCancerTypes = Arrays
+                .stream(SpecialCancerType.values())
+                .map(specialType ->
+                    allCancerTypes
+                        .stream()
+                        .filter(cancerType ->
+                            !StringUtils.isNullOrEmpty(cancerType.getMainType()) &&
+                            cancerType.getMainType().equals(specialType.getTumorType())
+                        )
+                        .findAny()
+                        .orElse(null)
+                )
+                .filter(cancerType -> cancerType != null)
+                .collect(Collectors.toList());
+            allOncoTreeTypes.removeAll(specialCancerTypes);
+
+            switch (specialCancerType) {
+                case ALL_SOLID_TUMORS:
+                    relevantCancerTypes.addAll(
+                        allOncoTreeTypes
+                            .stream()
+                            .filter(cancerType ->
+                                TumorForm.SOLID.equals(cancerType.getTumorForm()) || TumorForm.MIXED.equals(cancerType.getTumorForm())
+                            )
+                            .collect(Collectors.toSet())
+                    );
+                    break;
+                case ALL_LIQUID_TUMORS:
+                    relevantCancerTypes.addAll(
+                        allOncoTreeTypes
+                            .stream()
+                            .filter(cancerType ->
+                                TumorForm.LIQUID.equals(cancerType.getTumorForm()) || TumorForm.MIXED.equals(cancerType.getTumorForm())
+                            )
+                            .collect(Collectors.toSet())
+                    );
+                    break;
+                case ALL_TUMORS:
+                    relevantCancerTypes.addAll(allOncoTreeTypes);
+                    for (CancerType specialType : specialCancerTypes) {
+                        if (
+                            specialType.getCode().equals(SpecialCancerType.ALL_SOLID_TUMORS.name()) ||
+                            specialType.getCode().equals(SpecialCancerType.ALL_LIQUID_TUMORS.name())
+                        ) {
+                            relevantCancerTypes.add(specialType);
+                        }
+                    }
```

Comment:
Doesn't allOncoTreeTypes already include the special cancer type?

Reply:
I refactored so I don't use allOncoTreeTypes anymore